graph/telemetry/istio/appender/security_policy.go

```diff
@@ -195,6 +195,35 @@ func (a SecurityPolicyAppender) appendGraph(trafficMap graph.TrafficMap, namespa
 			query = fmt.Sprintf(`%s OR (%s)`, query, tcpReceivedQuery)
 		}
 	}
+	// If we are including ztunnel traffic we may need more TCP queries, because for ztunnel-to-sidecar traffic ztunnel
+	// will report as source telemetry. If this results in duplicate source and dest ztunnel telem, it should be OK, as
+	// the rate should be the same, and will just overwrite itself.
+	if a.Rates.Ambient == graph.AmbientTrafficTotal || a.Rates.Ambient == graph.AmbientTrafficZtunnel {
```

Comment:
I've though it would be useful for this waypoint code: https://github.com/kiali/kiali/pull/7756/files#diff-55173479d7c6bec43bb8b3b60d6b5d67bd408e9d3b8c70ea176c6b75e4eff0cfR317 
But it looks like it is not applicable here.

Reply:
I don't think so, but I do think we will see more corner cases. We always want to minimize the number of queries, so we have to be careful about what we add. One thing to notice in this appender is that we actually minimize the number of prometheus round trips, because we use OR expressions.  This is sort of a trick, but works as long as each sub-query has the same `by (...)` clause.  When that is true you can actually query different metrics all within the same prom query.